src/vos/evt_iter.c

```diff
@@ -550,6 +550,33 @@ int evt_iter_delete(daos_handle_t ih, struct evt_entry *ent)
 
 D_CASSERT(sizeof(((daos_anchor_t *)0)->da_buf) >= sizeof(struct evt_rect));
 
+int
+evt_iter_corrupt(daos_handle_t ih)
+{
+	struct evt_iterator	*iter;
+	struct evt_context	*tcx;
+	struct evt_trace	*trace;
+	struct evt_desc		*desc;
+	int			 rc;
+
+	tcx = evt_hdl2tcx(ih);
+	if (tcx == NULL)
+		return -DER_NO_HDL;
+
+	iter = &tcx->tc_iter;
+	rc = evt_iter_is_ready(iter);
+	if (rc != 0)
+		return rc;
+
+	trace = &tcx->tc_trace[tcx->tc_depth - 1];
+	desc = evt_node_desc_at(tcx, evt_off2node(tcx, trace->tr_node),
+				trace->tr_at);
+	D_DEBUG(DB_IO, "Setting record bio_addr flag to corrupted\n");
+	BIO_ADDR_SET_CORRUPTED(&desc->dc_ex_addr);
```

Comment:
I'm wondering why don't make it into a transaction to ensure the persistency (like what you did for dbtree_iter_corrupt()).